Data/models/DataController.swift

```diff
@@ -20,6 +20,37 @@ enum WriteContext {
 
 public class DataController: NSObject {
     private static let databaseName = "Brave.sqlite"
+    private static let modelName = "Model"
+    private let container: NSPersistentContainer
+    
+    // MARK: - Initialization
+    
+    /// Managed Object Model of the database stack.
+    /// Must be created only once, this is to prevent a bug when testing with in-memory store.
+    /// More info here https://stackoverflow.com/a/51857486.
+    /// Note: this might be not needed in Swift 5.1 or newer.
+    private static let model: NSManagedObjectModel = {
+        guard let modelURL = Bundle(for: DataController.self).url(forResource: modelName, withExtension: "momd") else {
+            fatalError("Error loading model from bundle")
+        }
+        guard let mom = NSManagedObjectModel(contentsOf: modelURL) else {
+            fatalError("Error initializing managed object model from: \(modelURL)")
+        }
+        
+        return mom
+    }()
+    
+    // Internal scope so unit tests can recreate the DB stack as needed.
+    override init() {
+        container = NSPersistentContainer(name: DataController.modelName,
```

Comment:
Previously `container` was a lazy property which did not guaranteed that it will be assigned only once(lazy is not atomic). This was happening sometimes in unit tests on background threads